docs/ibc/apps/ibcmodule.md

```diff
@@ -0,0 +1,350 @@
+<!--
+order: 2
+-->
+
+# Implement `IBCModule` interface and callbacks
+
+Learn how to implement the `IBCModule` interface and all of the callbacks it requires. {synopsis}
+
+The Cosmos SDK expects all IBC modules to implement the [`IBCModule`
+interface](https://github.com/cosmos/ibc-go/tree/main/modules/core/05-port/types/module.go). This interface contains all of the callbacks IBC expects modules to implement. The include callbacks related to channel handshake, opening and closing and packet callbacks (`OnRecvPacket`, `OnAcknowledgementPacket` and `OnTimeoutPacket`).
+
+```go
+// IBCModule implements the ICS26 interface for transfer given the transfer keeper.
+type IBCModule struct {
+	keeper keeper.Keeper
+}
+```
+
+Additionally, in the `module.go` file, add the following line:
+
+```go
+var (
+    _ module.AppModule      = AppModule{}
+    _ module.AppModuleBasic = AppModuleBasic{}
+    // Add this line
+    _ porttypes.IBCModule   = IBCModule{}
+)
+```
+
+## Pre-requisites Readings
+
+- [IBC Overview](../overview.md)) {prereq}
+- [IBC default integration](../integration.md) {prereq}
+
+## Channel handshake callbacks
+
+This section will describe the callbacks that are called during channel handshake execution.
+
+Here are the channel handshake callbacks that modules are expected to implement:
+
+> Note that some of the code below is _pseudo code_, indicating what actions need to happen but leaving it up to the developer to implement a custom implementation. E.g. the `checkArguments(args)` function.
+
+```go
+// Called by IBC Handler on MsgOpenInit
+func (im IBCModule) OnChanOpenInit(ctx sdk.Context,
+    order channeltypes.Order,
+    connectionHops []string,
+    portID string,
+    channelID string,
+    channelCap *capabilitytypes.Capability,
+    counterparty channeltypes.Counterparty,
+    version string,
+) (string,error) {
+    // ... do custom initialization logic
+
+    // Use above arguments to determine if we want to abort handshake
+    // Examples: Abort if order == UNORDERED,
+    // Abort if version is unsupported
+    if err := checkArguments(args); err != nil {
+        return "", err
+    }
+
+    if strings.TrimSpace(version) == "" {
+		version = types.Version
+	}
+
+	if version != types.Version {
+		return "", sdkerrors.Wrapf(types.ErrInvalidVersion, "got %s, expected %s", version, types.Version)
+	}
```

Comment:
this may not be true. Applications might generate some custom version that isn't static. Maybe replace with pseudo-code that checks the provided version for validity. ICS27 is an example of a version string that is dynamic per channel

Reply:
good catch, thanks!